bazel/bazel.go

```diff
@@ -79,6 +79,13 @@ func bazeliskNpmPath(ibazelBinPath string) (string, error) {
 				ext = ".exe"
 			}
 			name := strings.Join(append(prefix, nm, scope, "bazelisk", "bazelisk-"+bin+ext), "/")
+			_, err := os.Stat(name)
+			if err != nil {
```

Comment:
If the file exists but you can not stat it and it is at the end of the path .../node_modules/@bazel/ibazel/bin/<arch>/ibazel, why would continuing matter? Can this occur in practice?
But really, my question is realated the the other comment, which is why this is a loop in the first place.

Reply:
So I'm continuing if it doesn't exist? And returning the error if we fail to stat the file?

Reply:
Maybe I'm confused by the example.  Can the match at line 76 ever happen more than once?
It seems like once we find   .../node_modules/@bazel/ibazel/bin/bazilisk then the method should either return success or failure.

Reply:
@aiuto yes, the match can occur multiple times. It can happen once for each nested `node_modules`. Can you help me understand a bit better the situation where you think it will continue. Just to make things a little cleaner, here is the snippet embedded without comments:

https://github.com/bazelbuild/bazel-watcher/blob/56d637b7562dc74c1b3e5333d2dacd3a8301f64c/bazel/bazel.go#L73-L92

To me, the control flow appears to be:

1. Match on `node_modules/@bazel/ibazel/bin`
1. If `node_modules/@bazel/ibazel/bin/bazilisk` is found, return success. If there is an error return that error.

I'm terribly sorry if I'm being dense here, but I don't see another path through the loop.

Reply:
Ah... I understand now. The comments at line 66 made me thing this was a path to a single file, and not something that is more like a shell $PATH.  So I was looking at the loop the wrong way.
Maybe copying a live example into the comment when this code was originally written would have helped.